File: DSA/DSA06026 SẮP XẾP NỔI BỌT.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;

    for (int i = 0; i < n; i++)
    {
        bool check = false;
        for (int j = 0; j < n-1; j++)
        {
            if (A[j] > A[j+1])
            {
                swap(A[j], A[j+1]);
                check = true;
            }
        }
        
        if (!check)
        {
            break;
        }
        
        cout << "Buoc " << i+1 << ": ";
        for (auto &&k : A) cout << k << " ";
        cout << endl;
    }
    
    
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    submit();

    return 0;
}

File: DSA_python/DSA06026 SẮP XẾP NỔI BỌT.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))
    
    for i in range(n):
        check = False
        for j in range(n-1):
            if A[j] > A[j+1]:
                A[j], A[j+1] = A[j+1], A[j]
                check = True
        if check == False:
            break

        print(f"Buoc {i+1}:", " ".join(map(str, A)))


submit()
